### ISSUE TITLE
[SymbolReader] Make symbol server timeout configurable

### ISSUE THREAD

**Issue by suprak** [microsoft/perfview#1895]
https://github.com/microsoft/perfview/blob/766a08ca8373d273d55da8b07147e7068e4d5f58/src/TraceEvent/Symbols/SymbolReader.cs#L1143C17-L1144C33

Working with PerfView on a slower internet environment and public symbol server frequently times out and gets blocked from additional attempts to resolve, leading to broken symbols.

> FindSymbolFilePath: Time 27.2141736 sec.  Timeout of 25 seconds exceeded for https://msdl.microsoft.com/download/symbols.  Setting as dead server

`SymbolResolver` could have a property on it, say `ServerTimeout` or something.

**Comment by brianrob** [comment]
I agree that something like this is worthwhile.  At the same time, we probably ought to change the retry behavior to ensure that a failed request doesn't mark the server dead for minutes at a time.  This used to be OK, but often now this is overkill, and worse, in server processes, this is just wrong.

**Comment by brianrob** [comment]
Let's do this:

- Make the timeout configurable via a property on the SymbolReader.
- Let's not mark servers as dead anymore.  If we fail to get an individual file for whatever reason, subsequent calls to the same symbol server should still go through.
